extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -1,146 +1,165 @@
             respondException(res, ex);
         }
 
         if ((url != null) && !url.trim().equals("") && !url.trim().equals("url")) {
-            URL obj = new URL(url);
-            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
-            BufferedReader inr = new BufferedReader(new InputStreamReader(con.getInputStream()));
-            String inputLine;
-            StringBuffer response = new StringBuffer();
-
-            while ((inputLine = inr.readLine()) != null) {
-                    response.append(inputLine);
+            try {
+                format = getFormatFromUrl(parsers, url, in);
+            } catch (IOException e) {
+                respondException(res, e);
             }
-            inr.close();
-            in = new ByteArrayInputStream(response.toString().getBytes());
-        }
-        for (RDFParser parser: parsers){
-            try{
-                if( baseuri == null) {
-                    baseuri = "";
-                }
-                parser.parse(in, baseuri);
-                format = parser.getRDFFormat().getDefaultMIMEType();
-                break;
-            }catch(Exception e){
-                System.out.println(e.toString());    
-                continue;
-            }
-        }
-        if (format == null) {
-            format = "application/rdf+xml";
+        } else {
+            format = ParseFullFile(parsers, in, baseuri);
         }
-        
-        try{
+        try {
             res.setCharacterEncoding("UTF-8");
             res.setHeader("Content-Type", "application/json");
             JSONWriter writer = new JSONWriter(res.getWriter());
             writer.object();
-            writer.key("RDFFormat"); writer.value(format);
+            writer.key("RDFFormat");
+            writer.value(format);
             writer.endObject();
-        }catch(Exception e){
+        } catch (Exception e) {
             respondException(res, e);
         }
     }
+
+    private String ParseFullFile(RDFParser[] parsers, InputStream[] in, String baseuri) {
+        String format = null;
+        for (int i = 0; i < parsers.length; i++) {
+            try {
+                if (baseuri == null) {
+                    baseuri = "";
+                }
+                parsers[i].parse(in[i], baseuri);
+                format = parsers[i].getRDFFormat().getDefaultMIMEType();
+                break;
+            } catch (Exception e) {
+                continue;
+            }
+        }
+        return format;
+    }
+
+    private String getFormatFromUrl(RDFParser[] parsers, String url, InputStream[] in)
+            throws IOException, MalformedURLException {
+        StringBuffer response = new StringBuffer();
+        URL obj = new URL(url);
+
+        for (RDFParser parser : parsers) {
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+            con.addRequestProperty("Accept", parser.getRDFFormat().getDefaultMIMEType());
```

Comment:
There's no need to loop over the parsers one by one, you can generate the accept header values as I said [here](https://github.com/DTL-FAIRData/FAIRifier/pull/47#discussion_r145086937), that makes it quite a bit faster (especially for slow servers).